### PR TITLE
Prepare Cross User transactions

### PR DESCRIPTION
**Pull request by Glumli**

### CODE REVIEW

File: test/services/fhirServiceTest.js [hunk 1]
```diff
@@ -10,253 +10,180 @@ import documentRoutes from '../../src/routes/documentRoutes';
...
 	it('searchRecords succeeds', (done) => {
-		const fhirResponseList = [rawFhirResponse];
-		const searchRecordsStub = sinon.stub(documentRoutes, 'searchRecords')
-			.returnsPromise().resolves(fhirResponseList);
-
+		const secondUserId = 'userId2';
+		const secondTag = 'tag2';
 		const params = {
-			user_ids: ['user1', 'user2'],
+			user_ids: [testVariables.userId, secondUserId],
```
Comment (Harsimran1) [inline test/services/fhirServiceTest.js]: this looks inconsistent, you can add all the variables in the testVariable

File: test/services/fhirServiceTest.js [hunk 2]
```diff
@@ -10,253 +10,180 @@ import documentRoutes from '../../src/routes/documentRoutes';
...
-
-		fhirService.downloadFhirRecord('fakeRecordId').then((res) => {
-			expect(res.body).to.deep.equal(fhirObject);
-			expect(downloadFhirStub).to.be.calledOnce;
-			expect(downloadFhirStub).to.be.calledWith('fakeUserId');
-			expect(userServiceResolveUserStub).to.be.calledOnce;
-			downloadFhirStub.restore();
-			done();
-		});
+	it('downloadFhirRecord - Happy Path', (done) => {
```
Comment (Harsimran1) [inline test/services/fhirServiceTest.js]: do we have tests for sad paths?

Comment (Glumli) [reply test/services/fhirServiceTest.js]: We never had tests for anything else then the Happy path for downloading.

File: test/services/fhirServiceTest.js (hunk 1 above)
Comment (Glumli) [reply test/services/fhirServiceTest.js]: I didn't want to put it in there, as it is only user in one test. But I can put it in there

File: src/services/UserService.js [hunk 3]
```diff
@@ -10,104 +10,88 @@ class UserService {
...
 	}
 
-	getUser() {
+	isCurrentUser(userId) {
+		const currentUser = this.getCurrentUser();
+		if (!currentUser || currentUser.id !== userId) return false;
+		return true;
+	}
+
+	getInternalUser(userId, partial = false) {
```
Comment (Harsimran1) [inline src/services/UserService.js]: This name sounds weird to me. How about getUserById?

File: src/services/FhirService.js [hunk 4]
```diff
@@ -61,14 +61,13 @@ class FHIRService {
 	}
 
 
-	downloadFhirRecord(recordId) {
-		return documentRoutes.downloadRecord(UserService.getUserId(), recordId)
+	downloadFhirRecord(ownerId, recordId) {
```
Comment (Harsimran1) [inline src/services/FhirService.js]: All these methods will be breaking terra, need to change terra and documentation too before deployment.

Comment (Glumli) [reply src/services/FhirService.js]: I know

File: src/services/UserService.js (hunk 3 above)
Comment (Glumli) [reply src/services/UserService.js]: We somehow need to make a difference between the get User that is used internatlly and the one externally
getUserByIf would be fine for the external one

File: src/services/FhirService.js (hunk 4 above)
Comment (Glumli) [reply src/services/FhirService.js]: I actually forgot the documentation again, but I'm on the update of Terra

**Pull request merged**